services/terraformer/server/clusterBuilder/clusterBuilder.go

```diff
@@ -407,3 +415,58 @@ func copyCIDRsToMetadata(data *NodepoolsData) {
 		data.Metadata[fmt.Sprintf(subnetCidrKeyTemplate, np.Name)] = np.Metadata[subnetCidrKey].GetCidr()
 	}
 }
+
+// generateProviderTemplates generates only the `provider.tpl` templates so terraform can
+// destroy the infra if needed.
+func generateProviderTemplates(current, desired *pb.ClusterInfo, clusterID, directory, suffix string) error {
+	currentNodepools := utils.GroupNodepoolsByProvider(current)
```

Comment:
Do we need to check whether `current` or `desired` is not `nil` before calling this function?

Reply:
No, it will be checked in the getProvidersUsed func.

https://github.com/berops/claudie/blob/93402dd20ecfec0335716b3bb7e865f8e013a004/services/terraformer/server/provider/provider.go#L62-L65